src/unpack.cpp

```diff
@@ -2407,7 +2408,7 @@ asmjs::unpack(const uint8_t* packed, const char* cb_name, uint32_t unpacked_size
 
 #endif
 
-#ifdef EMSCRIPTEN
+#if defined(__EMSCRIPTEN__) && !defined(CHECKED_OUTPUT_SIZE)
```

Comment:
In other words, being an emscripten build doesn't mean one is running the client-side unpacker, it could also be the server-side packer.